### PR TITLE
Update with gem devise: Reset password, resend confirmation and update profile

### PR DESCRIPTION
**Pull request by TanBinh1601**
## Related Tickets
- [#44393](https://edu-redmine.sun-asterisk.vn/issues/44393)

## WHAT (optional)
- Reset password
- Send or resend confirmation activated account
- Update your profile

## HOW
- Using gem devise

## Evidence (Screenshot or Video)
Screenshot from 2021-12-17 10-06-00
Screenshot from 2021-12-17 10-06-09
Screenshot from 2021-12-17 10-06-28

### CODE REVIEW

**Comment by TanBinh1601** [comment]
ready

File: app/views/users/registrations/edit.html.erb [hunk 1]
```diff
@@ -1,43 +1,62 @@
...
+        <%= f.text_field :phone_number, autofocus: true, class: "form-control" %>
+      </div>
+
+      <div class="field">
+        <%= f.label t "user.new.address" %><br />
+        <%= f.text_field :address, autofocus: true, class: "form-control" %>
+      </div>
+
+      <% if devise_mapping.confirmable? && resource.pending_reconfirmation? %>
+        <div>Currently waiting confirmation for: <%= resource.unconfirmed_email %></div>
```
Comment (huongnt-2545) [inline app/views/users/registrations/edit.html.erb]: I18n

File: app/controllers/application_controller.rb [hunk 2]
```diff
@@ -2,9 +2,20 @@ class ApplicationController < ActionController::Base
...
+
+  before_action :configure_permitted_parameters, if: :devise_controller?
   include SessionsHelper
   include CartHelper
 
+  protected
+
+  def configure_permitted_parameters
+    devise_parameter_sanitizer.permit(:account_update) do |u|
+      u.permit :name, :email, :id_card, :phone_number, :address,
```
Comment (huongnt-2545) [inline app/controllers/application_controller.rb]: khai báo các field trog model User để sử dụng
vd:
```
class User
  USER_ATTRS = %w(name email ...).freeze
...
end
```

File: app/views/users/registrations/edit.html.erb (hunk 1 above)
Comment (TanBinh1601) [reply app/views/users/registrations/edit.html.erb]: vâng ạ

File: app/controllers/application_controller.rb (hunk 2 above)
Comment (TanBinh1601) [reply app/controllers/application_controller.rb]: vâng ạ

**Comment by TanBinh1601** [comment]
ready

**Review by ghost — COMMENTED** [review]
merged

**Pull request merged**